mountpoint-s3-client/tests/put_object.rs

```diff
@@ -277,6 +279,65 @@ async fn test_put_review(pass_review: bool) {
     }
 }
 
+fn write_with_storage_class_test<F>(creator_fn: F, storage_class: &str)
```

Comment:
In fact, these don't belong here. We just want to test that `put_object` is correctly setting the storage class (or not). No need for the file system.